helper/schema/resource_data.go

```diff
@@ -213,9 +212,11 @@ func (d *ResourceData) State() *terraform.InstanceState {
 		}
 
 		raw := d.get([]string{k}, source)
-		rawMap[k] = raw.Value
-		if raw.ValueProcessed != nil {
-			rawMap[k] = raw.ValueProcessed
+		if raw.Exists && !raw.Computed {
```

Comment:
To repeat: the `raw.Computed` only indicates the value is _NOT set and computed_. So what you're experiencing sounds like a bug, still. Let me see if I can add a test case for setting computed values.